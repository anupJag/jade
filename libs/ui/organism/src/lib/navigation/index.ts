export * from './navigation';
export * from './navigation.type';
