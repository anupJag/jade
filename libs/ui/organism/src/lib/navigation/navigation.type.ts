export type NavigationProps = {
  drawerHandler: Function;
  labels: {
    offersLabel: string;
  };
};
