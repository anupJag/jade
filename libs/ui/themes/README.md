# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
