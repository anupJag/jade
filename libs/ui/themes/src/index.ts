export * from './lib/defaultTheme';
export * from './lib/styled';
export * from './lib/themeProvider';
export { default as styled } from './lib/styled';
export * from './lib/GlobalStyles';
