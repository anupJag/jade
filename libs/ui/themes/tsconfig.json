{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
