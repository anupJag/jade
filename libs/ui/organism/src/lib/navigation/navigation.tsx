import React, { FC } from 'react';
import { GlobalLabel } from '@jade/ui/molecules';
import { NavigationProps } from './navigation.type';
import * as S from './navigation.styled';
import { HyperLink } from '@jade/ui/atoms';

export const Navigation: FC<NavigationProps> = ({ drawerHandler }) => {
  return (
    <S.Nav>
      <S.NavList>
        <S.NavItem>
          <HyperLink
            onClick={e => {
              e.preventDefault();
              drawerHandler();
            }}
            href="/categories"
            type="PrimaryNav"
            title="Categories"
            hasSubnav={true}>
            Categories
          </HyperLink>
        </S.NavItem>
        <S.NavItem>
          <HyperLink href="/offers" type="PrimaryNav" title="Offers" hasSubnav={false}>
            <GlobalLabel path="offersLabel" />
          </HyperLink>
        </S.NavItem>
        <S.NavItem>
          <HyperLink href="/help/faq" type="PrimaryNav" title="Help" hasSubnav={false}>
            Help
          </HyperLink>
        </S.NavItem>
      </S.NavList>
    </S.Nav>
  );
};
