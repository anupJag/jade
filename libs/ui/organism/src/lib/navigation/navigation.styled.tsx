import { styled } from '@jade/ui/themes';

export const Nav = styled.nav``;
export const NavList = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: flex;
`;
export const NavItem = styled.li`
  margin: 0 0 0 ${props => props.theme.space[8]};
  padding: 0;
`;
