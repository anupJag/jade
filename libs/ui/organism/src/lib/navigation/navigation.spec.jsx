import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { ThemeProvider } from '@jade/ui/themes';
import '@testing-library/jest-dom/extend-expect';

import { Navigation } from './navigation';

const NO_OF_NAVIATION_LINKS = 3;

describe('<Navigation />', () => {
  let NavigationComponent;
  const drawerHandlerMock = jest.fn();

  beforeEach(() => {
    NavigationComponent = render(
      <ThemeProvider>
        <Navigation drawerHandler={drawerHandlerMock} />
      </ThemeProvider>,
    );
  });

  test('should render correctly', () => {
    const { asFragment, getByText } = NavigationComponent;
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render all Navigation links', () => {
    const { container } = NavigationComponent;
    expect(container.querySelectorAll('a').length).toEqual(NO_OF_NAVIATION_LINKS);
  });

  describe('Link :', () => {
    let categoryEl;
    let container;

    beforeEach(() => {
      ({ container } = NavigationComponent);
      categoryEl = container.querySelector('a[title="Categories"]');
    });

    test('Categories should have sub naviation', () => {
      const isLinkHasSubNav = categoryEl.hasChildNodes();
      expect(isLinkHasSubNav).toBeTruthy();
    });

    test('Categories should call handler function on click of link', () => {
      fireEvent.click(categoryEl);
      expect(drawerHandlerMock).toHaveBeenCalledTimes(1);
    });

    test('Categories should have href & title values', () => {
      const isHrefProvided = categoryEl.getAttribute('href');
      const isTitleProvided = categoryEl.getAttribute('title');
      expect(isHrefProvided).toBeTruthy();
      expect(isTitleProvided).toBeTruthy();
    });

    test('Offers should have href & title values', () => {
      const offersEl = container.querySelector('a[title="Offers"]');
      const isHrefProvided = offersEl.getAttribute('href');
      const isTitleProvided = offersEl.getAttribute('title');
      expect(isHrefProvided).toBeTruthy();
      expect(isTitleProvided).toBeTruthy();
    });

    test('Help should have href & title values', () => {
      const helpEl = container.querySelector('a[title="Help"]');
      const isHrefProvided = helpEl.getAttribute('href');
      const isTitleProvided = helpEl.getAttribute('title');
      expect(isHrefProvided).toBeTruthy();
      expect(isTitleProvided).toBeTruthy();
    });
  });
});
