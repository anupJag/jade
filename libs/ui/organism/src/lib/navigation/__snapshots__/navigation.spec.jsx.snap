// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render correctly 1`] = `
<DocumentFragment>
  <nav
    class="css-0"
  >
    <ul
      class="css-1q0n0ua"
    >
      <li
        class="css-1i54o6h"
      >
        <a
          class="css-mzqm6m"
          href="/categories"
          title="Categories"
        >
          Categories 
          <svg
            aria-hidden="true"
            class="css-1im46kq"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="currentColor"
            />
          </svg>
        </a>
      </li>
      <li
        class="css-1i54o6h"
      >
        <a
          class="css-mzqm6m"
          href="/offers"
          title="Offers"
        >
           
        </a>
      </li>
      <li
        class="css-1i54o6h"
      >
        <a
          class="css-mzqm6m"
          href="/help/faq"
          title="Help"
        >
          Help 
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
